Adds math tests for transforms, rotations, and conversions (#103)

jtigue-bdai opened the pull request:
# Description

this PR adds tests for:

- scale_transform
- unscale_transform
- saturate
- normalize
- copysign
- convert_quat
- quat_conjugate
- quat_from_euler_xyz
- quat_from_matrix
- euler_xyz_from_quat
- matrix_from_euler
- quat_from_angle_axis
- axis_angle_from_quat
- skew_symmetric_matrix
- combine_transform
- subtract_transform
- compute_pose_error

Fixes # (issue)

## Type of change

<!-- As you go through the list, delete the ones that are not applicable. -->

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [ ] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

<!--
As you go through the checklist above, you can mark something as done by putting an x character in it

For example,
- [x] I have done this task
- [ ] I have not done this task
-->

jtigue-bdai commented on source/isaaclab/test/utils/test_math.py:
```diff
@@ -35,6 +35,105 @@
+@pytest.mark.parametrize("device", ("cpu", "cuda:0"))
+@pytest.mark.parametrize("size", ((5, 4, 3), (10, 2)))
+def test_normalize(device, size):
+    """Test normalize."""
```
```suggestion
    """Test normalize of a tensor along its last dimension and check the norm of that dimension is close to 1.0."""
```

jtigue-bdai commented on source/isaaclab/test/utils/test_math.py:
```diff
@@ -257,6 +356,82 @@ def test_convention_converter(device):
+@pytest.mark.parametrize("device", ("cpu", "cuda:0"))
+@pytest.mark.parametrize("size", ((10, 4), (5, 3, 4)))
+def test_convert_quat(device, size):
+    """Test convert_quat."""
```
```suggestion
    """Test convert_quat from xyzw to wxyz and back to xyzw and verify the correct rolling of the tensor. Also check the correct exceptions are raised for bad inputs for the quaternion and the 'to'."""
```

jtigue-bdai commented on source/isaaclab/test/utils/test_math.py:
```diff
@@ -257,6 +356,82 @@ def test_convention_converter(device):
+
+@pytest.mark.parametrize("device", ("cpu", "cuda:0"))
+def test_quat_conjugate_and_inv(device):
+    """Test quat_conjugate and quat_inv."""
```
```suggestion
    """Test quat_conjugate by checking the sign of the imaginary part changes but the magnitudes stay the same."""
```

jtigue-bdai commented on source/isaaclab/test/utils/test_math.py:
```diff
@@ -462,6 +687,107 @@ def test_quat_box_minus_and_quat_box_plus(device):
+@pytest.mark.parametrize("t12_inputs", ["True", "False"])
+@pytest.mark.parametrize("q12_inputs", ["True", "False"])
+def test_combine_frame_transforms(device, t12_inputs, q12_inputs):
+    """Test combine_frame_transforms."""
```
```suggestion
    """Test combine_frame_transforms such that inputs for delta translation and delta rotation can be None or specified."""
```

ooctipus reviewed (approved):
This is looking really good, Thanks James!

jtigue-bdai commented:
> > > looks like a couple of the tests are failing in CI, is that expected?
> > 
> > 
> > Nope not expected. Let me take a look.
> 
> @kellyguo11 fixed the issue. it was the compute_pose_error changes. Once I reverted those and fixed the test it seemed to work on my machine.

Tests pass except `test_tiled_camera`

ooctipus commented:
Some tests in test_tiled_camera is a bit finicky because the rendered pixel are inherently stochastic in tiled camera, we fix some of it in the upcoming commits, but I think it should be fine. @kellyguo11
